bot/cogs/help_channels.py

```diff
@@ -183,15 +185,33 @@ def create_name_queue(self) -> deque:
         log.trace("Populating the name queue with names.")
         return deque(available_names)
 
+    async def dormant_check(self, ctx: commands.Context) -> bool:
+        """Return True if the user started the help channel session or passes the role check."""
+        if self.help_channel_users.get(ctx.channel) == ctx.author:
+            log.trace(f"{ctx.author} started the help session, passing the check for dormant.")
+            return True
+
+        log.trace(f"{ctx.author} did not start the help session, checking roles.")
+        return with_role_check(ctx, *constants.HelpChannels.cmd_whitelist)
+
     @commands.command(name="dormant", aliases=["close"], enabled=False)
-    @with_role(*constants.HelpChannels.cmd_whitelist)
     async def dormant_command(self, ctx: commands.Context) -> None:
-        """Make the current in-use help channel dormant."""
-        log.trace("dormant command invoked; checking if the channel is in-use.")
+        """
+        Make the current in-use help channel dormant.
 
+        Make the channel dormant if the user passes the `dormant_check`,
+        delete the message that invoked this,
+        and reset the send permissions cooldown for the user who started the session.
+        """
+        log.trace("dormant command invoked; checking if the channel is in-use.")
         if ctx.channel.category == self.in_use_category:
-            self.cancel_task(ctx.channel.id)
-            await self.move_to_dormant(ctx.channel)
+            if await self.dormant_check(ctx):
+                self.cancel_task(ctx.channel.id)
+                await self.move_to_dormant(ctx.channel)
+                with suppress(discord.errors.NotFound):
+                    await ctx.message.delete()
+                    log.trace("Deleting dormant invokation message.")
```

Comment:
It makes more sense for the log to go before the delete given the way you wrote the message.